### PR TITLE
feat: use kubernetes informers to notify when pods enter/exit the clu…

### PR DESCRIPTION
**Pull request by kuzmik**
…ster

This replaces the pulling data from the k8s api every 10s.

### CODE REVIEW

**Comment by kuzmik** [comment]
I had to pull the The previous version of this due to a strange issue with how proxysql processes commands. The idiomatic way of doing sql queries with arguments in Go just... doesn't work with the proxysql interface. So I went back to `fmt.Sprintf`

File: Makefile [hunk 1]
```diff
@@ -33,7 +33,7 @@ run: build
 	@./$(TARGET)
 
 docker: clean lint
-	@docker build -f build/dev.Dockerfile . -t persona-id/proxysql-agent:latest
+	@docker build -f build/dev.Dockerfile -t persona-id/proxysql-agent:latest .
```
Comment (derektamsen) [inline Makefile:36]: Is the docker make target for local testing? If at some point you would like to push to the registry the image should be tagged as `ghcr.io/persona-id/proxysql-agent:latest`. The current tag will default to docker hub.

**Review by derektamsen — APPROVED** [review]
Yay! More tests and you are using a k8s watcher to get pod updates instead of polling the api!

File: Makefile (hunk 1 above)
Comment (kuzmik) [reply Makefile:36]: ah yeah the makefile is pretty much only for testing locally, that's all i use it for. we have goreleaser setup in this repo to do the actual releases!

**Pull request merged**